gateway/src/main/java/io/zeebe/gateway/impl/probes/health/PartitionLeaderAwarenessHealthIndicator.java

```diff
@@ -20,20 +21,21 @@
  */
 public class PartitionLeaderAwarenessHealthIndicator implements HealthIndicator {
 
-  private final Supplier<BrokerClusterState> clusterStateSupplier;
+  private final Supplier<Optional<BrokerClusterState>> clusterStateSupplier;
 
   public PartitionLeaderAwarenessHealthIndicator(
-      final Supplier<BrokerClusterState> clusterStateSupplier) {
+      final Supplier<Optional<BrokerClusterState>> clusterStateSupplier) {
     this.clusterStateSupplier = requireNonNull(clusterStateSupplier);
   }
 
   @Override
   public Health health() {
-    final var clusterState = clusterStateSupplier.get();
+    final var optClusterState = clusterStateSupplier.get();
 
-    if (clusterState == null) {
+    if (optClusterState.isEmpty()) {
       return Health.down().build();
     } else {
+      final var clusterState = optClusterState.get();
```

Comment:
just and idea: This can probably then rewritten like:

```java
if (optClusterState.map(BrokerClusterState::getPartitions)
     .stream()
     .filter(index -> {
        return clusterState.getLeaderForPartition(index) != BrokerClusterState.NODE_ID_NULL;
       })
      .findAny()
      .isPresent())
{
 return Health.up().build();
}
else {
return Health.down().build();
}
```